src/test/javascript/spec/component/learning-paths/components/learning-path-lecture-unit.component.spec.ts

```diff
@@ -110,4 +118,25 @@ describe('LearningPathLectureUnitComponent', () => {
         expect(setIsLoadingSpy).toHaveBeenCalledWith(true);
         expect(setIsLoadingSpy).toHaveBeenCalledWith(false);
     });
+
+    it('should show discussion section when communication is enabled', async () => {
+        fixture.detectChanges();
+        await fixture.whenStable();
+        fixture.detectChanges();
+
+        const discussionSection = fixture.nativeElement.querySelector('jhi-discussion-section');
+        expect(discussionSection).toBeTruthy();
+    });
+
+    it('should not show discussion section when communication is disabled', async () => {
+        const lecture = { ...lectureUnit.lecture, course: { courseInformationSharingConfiguration: CourseInformationSharingConfiguration.DISABLED } };
+        getLectureUnitByIdSpy.mockReturnValue(of({ ...lectureUnit, lecture }));
+
+        fixture.detectChanges();
+        await fixture.whenStable();
+        fixture.detectChanges();
```

Comment:
why do we need 2 detect changes?

Reply:
true here we can omit it but the method before we need it twice.